index.d.ts

```diff
@@ -5,11 +5,13 @@ declare module "effection" {
 
   export interface Sequence extends Generator<Operation, any, any> {}
 
-  export interface Execution<T = any> {
+  export interface Execution<T = any> extends PromiseLike<any> {
```

Comment:
Eventually, it would be nice to be able to specify the type of this promise, since the argument to fork is *usually* a generator, and we know the return value of a generator, we should actually be able to thread this value through, I think?